File: Greedy/C++/Drones.cpp
#include <bits/stdc++.h>
using namespace std;

struct Drone
{
    int pos;
    int dis;
};

bool cmp(Drone d1, Drone d2)
{
    return d1.dis < d2.dis;
}

void Solution(vector<Drone> drones)
{
    sort(drones.begin(), drones.end(), cmp);
    int second = 0;
    for (int i = 0; i < drones.size(); i++)
    {
        if (second == drones[i].dis)
        {
            cout << -1;
            return;
        }
        second++;
    }
    for (int i = 0; i < drones.size(); i++)
    {
        cout << drones[i].pos << " ";
    }
}

int main()
{
    int n;
    cin >> n;
    vector<Drone> drones;
    for (int i = 0; i < n; i++)
    {
        int x, y;
        cin >> x >> y;
        Drone temp;
        temp.pos = i + 1;
        // Chebyshev Distance
        temp.dis = max(abs(x), abs(y));
        drones.push_back(temp);
    }
    Solution(drones);
    return 0;
}

File: Greedy/Python/Drones.py
# Chebyshev Distance
# TIME LIMIT EXCEEDED
def solution(drones):
    drones.sort(key=lambda x: x[2])
    second = 0
    for i in range(n):
        if second == drones[i][2]:
            print(-1)
            return
        second += 1
    for j in range(n):
        print(drones[j][1], end=" ")


if __name__ == "__main__":
    n = int(input())
    drones = []
    for i in range(n):
        x, y = map(int, input().split())
        drones.append([(x, y), i + 1, max(abs(x), abs(y))])
    solution(drones)
